client/checkout/utils/upe.js

```diff
@@ -300,3 +300,64 @@ export const getStripeElementOptions = (
 
 	return options;
 };
+
+/**
+ * Converts form fields object into Stripe `billing_details` object.
+ *
+ * @param {Object} fields Object mapping checkout billing fields to values.
+ * @return {Object} Stripe formatted `billing_details` object.
+ */
+export const getBillingDetails = ( fields ) => {
+	return {
+		name:
+			`${ fields.billing_first_name } ${ fields.billing_last_name }`.trim() ||
+			'-',
+		email:
+			'string' === typeof fields.billing_email
+				? fields.billing_email.trim()
+				: '-',
+		phone: fields.billing_phone || '-',
+		address: {
+			country: fields.billing_country || '-',
+			line1: fields.billing_address_1 || '-',
+			line2: fields.billing_address_2 || '-',
+			city: fields.billing_city || '-',
+			state: fields.billing_state || '-',
+			postal_code: fields.billing_postcode || '-',
+		},
+	};
+};
+
+/**
+ * Converts form fields object into Stripe `shipping` object.
+ *
+ * @param {Object} fields Object mapping checkout shipping fields to values.
+ * @return {Object} Stripe formatted `shipping` object.
+ */
+export const getShippingDetails = ( fields ) => {
+	// Shipping address is needed by Afterpay. If available, use shipping address, else fallback to billing address.
+	if (
+		fields.ship_to_different_address &&
+		'1' === fields.ship_to_different_address
+	) {
+		return {
+			name:
+				`${ fields.shipping_first_name } ${ fields.shipping_last_name }`.trim() ||
+				'-',
+			address: {
+				country: fields.shipping_country || '-',
+				line1: fields.shipping_address_1 || '-',
+				line2: fields.shipping_address_2 || '-',
+				city: fields.shipping_city || '-',
+				state: fields.shipping_state || '-',
+				postal_code: fields.shipping_postcode || '-',
+			},
+		};
+	}
+
+	const billingAsShippingAddress = getBillingDetails( fields );
```

Comment:
Is delete working with constant-variables?

Reply:
It's working here, since we are deleting the properties within the const object. Props to ESLint :D that gave suggestion to replace `let` with `const`, since there is no reassignment of the whole object.

Reply:
Gotcha, thanks!